act/registry.go

```diff
@@ -26,6 +28,10 @@ type Registry struct {
 	// Default timeout for running actions
 	DefaultActionTimeout time.Duration
 
+	// TaskPublisher is the publisher for async actions.
+	// if not given, will invoke simple goroutine to run async actions
+	TaskPublisher *Publisher
```

Comment:
You were using an interface here, which causes the pointer not to work as expected, and I fixed it by using to the concrete type.